### ISSUE TITLE
`context` variable is not reported as undefined

### ISSUE THREAD

**Issue by jugarrit** [eslint/eslint#5447]
**What version of ESLint are you using?**
v2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**What did you do? Please include the actual source code causing the issue.**
I used a variable called `context` without defining it.

```js
class MyClass {
    constructor() {
        console.log(thisworks);
        console.log('this does not :: ', context);
    }
}
```

```shell
/Users/julian/src/smooch/text.js
  3:21  error  'thisworks' is not defined  no-undef
✖ 1 problem (1 error, 0 warnings)
```

**What did you expect to happen?**
I expected eslint to detect an error on the `context` variable.

**What actually happened? Please include the actual, raw output from ESLint.**
eslint did not detect an error on the `context` variable.

**Comment by michaelficarra** [comment]
My guess is that you're using the `mocha` environment, since [it defines a `context` global](https://github.com/sindresorhus/globals/blob/1e9ebc39828b92bd5c8ec7dc7bb07d62f2fb0153/globals.json#L857).

**Comment by jugarrit** [comment]
That's it! Thanks :)
